metadata-ingestion/src/datahub/specific/dataset.py

```diff
@@ -142,6 +154,18 @@ def remove_upstream_lineage(
         return self
 
     def set_upstream_lineages(self, upstreams: List[Upstream]) -> "DatasetPatchBuilder":
```

Comment:
does it make sense for these to be auto-set by the add / set methods? or should this be set by whatever source calls this method?

e.g. I would think that this should be set by the redshift source